File: Neetcode/RevPolish.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        op = set(["+", "-", "*", "/"])
        for t in tokens:
            if t not in op:
                stack.append(int(t))
            else:
                numR = int(stack.pop())
                numL = int(stack.pop())
                if t == "+":
                    stack.append(numL + numR)
                elif t == "-":
                    stack.append(numL - numR)
                elif t == "*":
                    stack.append(numL * numR)
                elif t == "/":
                    stack.append(int(numL / numR))
        return stack[-1]

File: Neetcode/RevPolish.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        int r, l;
        for(string c: tokens){
            if (c == "+"){
                r = s.top();
                s.pop();
                l = s.top();
                s.pop();
                s.push(l + r);
            }
            else if (c == "-"){
                r = s.top();
                s.pop();
                l = s.top();
                s.pop();
                s.push(l - r);
            }
            else if (c == "*"){
                r = s.top();
                s.pop();
                l = s.top();
                s.pop();
                s.push(l * r);
            }
            else if (c == "/"){
                r = s.top();
                s.pop();
                l = s.top();
                s.pop();
                s.push(l / r);
            }
            else{
                s.push(stoi(c));
            }
        }
        return s.top();
    }
};
